### PR TITLE
0.6.2 updates

### PR DESCRIPTION
**Pull request by tersec**

### CODE REVIEW

**Comment by tersec** [comment]
The three commit build failures are all AppVeyor timing out connecting to GitHub on three different repos, so transient networking issues that eventually didn't occur:

https://ci.appveyor.com/project/nimbus/nim-beacon-chain/builds/24990062/job/9srtl9g7c9fs4i7w has:
```
Downloading https://github.com/status-im/nim-ranges using git
Error: unhandled exception: Unable to query remote tags for https://github.com/status-im/nim-ranges. Git returned: fatal: unable to access 'https://github.com/status-im/nim-ranges/': Failed to connect to github.com port 443: Timed out
 [OSError]
```

https://ci.appveyor.com/project/nimbus/nim-beacon-chain/builds/24990347/job/a6ikfdvvvylepuv1 has:
```
     Error: Execution failed with exit code 128
        ... Command: git clone --recursive --depth 1 https://github.com/status-im/nim-json-rpc C:\Users\appveyor\AppData\Local\Temp\1\nimble_2404\githubcom_statusimnimjsonrpc
        ... Output: Cloning into 'C:\Users\appveyor\AppData\Local\Temp\1\nimble_2404\githubcom_statusimnimjsonrpc'...
        ... fatal: unable to access 'https://github.com/status-im/nim-json-rpc/': Failed to connect to github.com port 443: Timed out
```

https://ci.appveyor.com/project/nimbus/nim-beacon-chain/builds/24990696/job/a5heyucbathjdl4c has:
```
Downloading https://github.com/status-im/nim-serialization using git
Error: unhandled exception: Unable to query remote tags for https://github.com/status-im/nim-serialization. Git returned: fatal: unable to access 'https://github.com/status-im/nim-serialization/': Failed to connect to github.com port 443: Timed out
 [OSError]
```

File: beacon_chain/state_transition.nim [hunk 1]
```diff
@@ -976,25 +910,42 @@ func process_slashings(state: var BeaconState) =
...
-    (next_epoch + ACTIVATION_EXIT_DELAY) mod LATEST_ACTIVE_INDEX_ROOTS_LENGTH
-  state.latest_active_index_roots[index_root_position] =
-    hash_tree_root(get_active_validator_indices(
-      state, next_epoch + ACTIVATION_EXIT_DELAY)
-  )
+  # Reset eth1 data votes
+  if (state.slot + 1) mod SLOTS_PER_ETH1_VOTING_PERIOD == 0:
+    state.eth1_data_votes = @[]
+
+  # Update effective balances with hysteresis
```
Comment (arnetheduck) [inline beacon_chain/state_transition.nim]: fancy word.

File: beacon_chain/spec/validator.nim [hunk 2]
```diff
@@ -291,6 +171,47 @@ func get_crosslink_committee*(state: BeaconState, epoch: Epoch, shard: Shard):
...
+      get_crosslink_committee(state, epoch, shard),
+      shard
+    )
+
+iterator get_crosslink_committees_at_slot_cached*(
+  state: BeaconState, slot: Slot|uint64, cache: var StateCache):
+    CrosslinkCommittee =
+  let key = (slot_to_epoch(slot).uint64, false)
+  if key in cache.crosslink_committee_cache:
+    for v in cache.crosslink_committee_cache[key]: yield v
```
Comment (arnetheduck) [inline beacon_chain/spec/validator.nim]: return?

Comment (tersec) [reply beacon_chain/spec/validator.nim]: I changed as little regarding `get_crosslink_committees_at_slot_cached` as feasible, only removed the `registry_change` parameters. Didn't have `return` before. I just moved `get_crosslink_committees_at_slot_cached` and `get_crosslink_committees_at_slot` to after `compute_committee`.

This is a legacy interface now, and the fixes should be in the form of removing call sites.

Comment (arnetheduck) [reply beacon_chain/spec/validator.nim]: sure, looks like a pretty expensive bug, is all.. that it hasn't been caught in tests is.. intriguing.

**Pull request merged**